### DISCUSSION TITLE
Pillow and AWS Lambda

### DISCUSSION THREAD

**Discussion by Tsingis** [python-pillow/Pillow#7117] [Q&A] [answered]
Hi been receiving error for a while now while running some simple image manipulations on AWS Lamba and python3.9/3.10.

```ImportError: /lib64/libz.so.1: version `ZLIB_1.2.9' not found (required by Pillow.libs/libpng16-cb42a3c6.so.16.39.0)```

Seems to have something to do with https://pillow.readthedocs.io/en/stable/reference/ImageFont.html#PIL.ImageFont.truetype according to stack trace.

No issues on my personal Windows10/11 or Github workflows with ubuntu-latest (22.04) or windows-latest (Server 2022).

What should I do?

**Comment by nulano** [comment]
libpng is only used by libfreetype, which is only used by the ImageFont module, so it makes sense that you are seeing it there in the stack trace.

How did you install Pillow? Are you using the latest version?

From what I recall, the system libraries provided on AWS Lambda tend to be quite old, so I would assume that pip installs with `Pillow-9.5.0-...-manylinux_2_17_x86_64.manylinux2014_x86_64.whl`, so the https://github.com/python-pillow/pillow-wheels/issues/255 issue may be related to this.

**Comment by Tsingis** [comment]
I have been using [Serverless Framework ](https://www.serverless.com/framework/docs/providers/aws/guide/serverless.yml) for deployment and using `serverless-python-requirements` plugin to specify `requirements.txt` with `Pillow==9.5.0`

AWS seems to run Lambdas on Amazon Linux 2. https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html

Could also be related to this https://github.com/serverless/serverless/pull/11922 however had issues with 3.9 too.

**Comment by Tsingis** [comment, marked as answer]
It seems that as I was deploying locally via serverless from windows machine with parameter `dockerizePip: non-linux` it was using `public.ecr.aws/sam/build-python3.10:latest-x86-64` under the hood for non-linux machines. Switched my build automation to use this as well and everything works.
